storage: TestReplicateRemovedNodeDisruptiveElection is disabled

The following test appears to have failed:

[#20977](https://circleci.com/gh/cockroachdb/cockroach/20977):

```
I160803 19:43:02.089971 kv/txn_coord_sender.go:246  txn coordinator: 0.00 txn/sec, 100.00/0.00/0.00/0.00 %cmmt/cmmt1pc/abrt/abnd, 0s/0s/0s avg/σ/max duration, 0.0/0.0/0 avg/σ/max restarts (0 samples)
I160803 19:43:06.899636 kv/txn_coord_sender.go:246  txn coordinator: 0.00 txn/sec, 0.00/0.00/0.00/0.00 %cmmt/cmmt1pc/abrt/abnd, 0s/0s/0s avg/σ/max duration, 0.0/0.0/0 avg/σ/max restarts (0 samples)
I160803 19:43:06.927221 kv/txn_coord_sender.go:246  txn coordinator: 0.00 txn/sec, 0.00/0.00/0.00/0.00 %cmmt/cmmt1pc/abrt/abnd, 0s/0s/0s avg/σ/max duration, 0.0/0.0/0 avg/σ/max restarts (0 samples)
I160803 19:43:06.928929 kv/txn_coord_sender.go:246  txn coordinator: 0.00 txn/sec, 0.00/0.00/0.00/0.00 %cmmt/cmmt1pc/abrt/abnd, 0s/0s/0s avg/σ/max duration, 0.0/0.0/0 avg/σ/max restarts (0 samples)
I160803 19:43:07.091130 kv/txn_coord_sender.go:246  txn coordinator: 0.00 txn/sec, 100.00/0.00/0.00/0.00 %cmmt/cmmt1pc/abrt/abnd, 0s/0s/0s avg/σ/max duration, 0.0/0.0/0 avg/σ/max restarts (0 samples)
panic: test timed out after 5m0s

goroutine 38355 [running]:
panic(0x1a3e160, 0xc421883ae0)
    /usr/local/go/src/runtime/panic.go:500 +0x1ae
testing.startAlarm.func1()
    /usr/local/go/src/testing/testing.go:918 +0x14e
created by time.goFunc
    /usr/local/go/src/time/sleep.go:154 +0x78

goroutine 1 [chan receive, 4 minutes]:
testing.(*T).Run(0xc420258240, 0x1bfd520, 0x2a, 0x1e73ae8, 0x95b101)
    /usr/local/go/src/testing/testing.go:647 +0x56e
testing.RunTests.func1(0xc420258240)
    /usr/local/go/src/testing/testing.go:793 +0xba
testing.tRunner(0xc420258240, 0xc420477b38)
    /usr/local/go/src/testing/testing.go:610 +0xca
testing.RunTests(0x1e74cd8, 0x2715ce0, 0x12b, 0x12b, 0xc42022a7c8)
    /usr/local/go/src/testing/testing.go:799 +0x4b6
testing.(*M).Run(0xc420477ef0, 0xc420253ce0)
    /usr/local/go/src/testing/testing.go:743 +0x130
github.com/cockroachdb/cockroach/storage_test.TestMain(0xc420477ef0)
    /go/src/github.com/cockroachdb/cockroach/storage/main_test.go:57 +0x27b
main.main()
    github.com/cockroachdb/cockroach/storage/_test/_testmain.go:660 +0x1b2

goroutine 17 [syscall, 4 minutes, locked to thread]:
runtime.goexit()
    /usr/local/go/src/runtime/asm_amd64.s:2086 +0x1

goroutine 5 [chan receive]:
github.com/cockroachdb/cockroach/util/log.(*loggingT).flushDaemon(0x2a4fe60)
    /go/src/github.com/cockroachdb/cockroach/util/log/clog.go:1015 +0x85
created by github.com/cockroachdb/cockroach/util/log.init.1
    /go/src/github.com/cockroachdb/cockroach/util/log/clog.go:598 +0xc3

goroutine 19661 [select]:
google.golang.org/grpc/transport.(*recvBufferReader).Read(0xc422c94200, 0xc422e8a230, 0x5, 0x5, 0x0, 0x0, 0x0)
    /go/src/google.golang.org/grpc/transport/transport.go:141 +0x898
google.golang.org/grpc/transport.(*Stream).Read(0xc422b52690, 0xc422e8a230, 0x5, 0x5, 0x608fb7, 0xc4218e6970, 0xc4218e6954)
    /go/src/google.golang.org/grpc/transport/transport.go:314 +0x79
io.ReadAtLeast(0x271f800, 0xc422b52690, 0xc422e8a230, 0x5, 0x5, 0x5, 0xc420777840, 0x60933a, 0xc420076000)
    /usr/local/go/src/io/io.go:307 +0xb2
io.ReadFull(0x271f800, 0xc422b52690, 0xc422e8a230, 0x5, 0x5, 0xc4ffffffff, 0x0, 0xc4218e6900)
    /usr/local/go/src/io/io.go:325 +0x73
google.golang.org/grpc.(*parser).recvMsg(0xc422e8a220, 0x7fffffff, 0x5d14a5, 0xc422d4e298, 0x0, 0x0, 0x6, 0x0)
    /go/src/google.golang.org/grpc/rpc_util.go:231 +0xa4
--
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc420c52000, 0xc422c94c00)
    /go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:187 +0x8b
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
    /go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:188 +0x74

goroutine 19540 [select, 4 minutes]:
github.com/cockroachdb/cockroach/storage.(*baseQueue).processLoop.func1()
    /go/src/github.com/cockroachdb/cockroach/storage/queue.go:402 +0x366
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc420c52120, 0xc422f37300)
    /go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:187 +0x8b
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
    /go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:188 +0x74

goroutine 19549 [select]:
github.com/cockroachdb/cockroach/gossip.(*client).gossip(0xc421f70820, 0x7f8a4156ef30, 0xc422ed7fc0, 0xc4216d6420, 0x271c2c0, 0xc420028150, 0xc420c52000, 0x0, 0x0)
    /go/src/github.com/cockroachdb/cockroach/gossip/client.go:278 +0x6df
github.com/cockroachdb/cockroach/gossip.(*client).start.func1()
    /go/src/github.com/cockroachdb/cockroach/gossip/client.go:93 +0x2e9
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc420c52000, 0xc422ed7dc0)
    /go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:187 +0x8b
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
    /go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:188 +0x74

goroutine 19619 [select, 4 minutes]:
github.com/cockroachdb/cockroach/gossip.(*Gossip).bootstrap.func1()
    /go/src/github.com/cockroachdb/cockroach/gossip/gossip.go:871 +0x355
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc420c52000, 0xc422cf52c0)
    /go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:187 +0x8b
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
    /go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:188 +0x74

goroutine 19716 [IO wait]:
net.runtime_pollWait(0x7f8a415298c0, 0x72, 0x5de873)
    /usr/local/go/src/runtime/netpoll.go:160 +0x5e
net.(*pollDesc).wait(0xc421d3bd40, 0x72, 0x5db128, 0x1ab1240)
    /usr/local/go/src/net/fd_poll_runtime.go:73 +0x5b
net.(*pollDesc).waitRead(0xc421d3bd40, 0x2723240, 0xc420012098)
    /usr/local/go/src/net/fd_poll_runtime.go:78 +0x42
net.(*netFD).Read(0xc421d3bce0, 0xc422f94000, 0x8000, 0x8000, 0x0, 0x2723240, 0xc420012098)
    /usr/local/go/src/net/fd_unix.go:220 +0x1d9
net.(*conn).Read(0xc42012c720, 0xc422f94000, 0x8000, 0x8000, 0x2, 0x0, 0x0)
    /usr/local/go/src/net/net.go:173 +0x97
--
google.golang.org/grpc/transport.(*http2Client).reader(0xc4216dd900)
    /go/src/google.golang.org/grpc/transport/http2_client.go:889 +0x157
created by google.golang.org/grpc/transport.newHTTP2Client
    /go/src/google.golang.org/grpc/transport/http2_client.go:171 +0xbaf

goroutine 19638 [select]:
github.com/cockroachdb/cockroach/storage.(*Store).startGossip.func2()
    /go/src/github.com/cockroachdb/cockroach/storage/store.go:1227 +0x2e9
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc420c52ab0, 0xc422ed2b80)
    /go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:187 +0x8b
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
    /go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:188 +0x74

goroutine 19949 [select]:
google.golang.org/grpc/transport.(*recvBufferReader).Read(0xc421c1cdc0, 0xc42296e490, 0x5, 0x5, 0x0, 0x0, 0x0)
    /go/src/google.golang.org/grpc/transport/transport.go:141 +0x898
google.golang.org/grpc/transport.(*Stream).Read(0xc4209eef00, 0xc42296e490, 0x5, 0x5, 0x0, 0xc420cd8180, 0xc420cd8140)
    /go/src/google.golang.org/grpc/transport/transport.go:314 +0x79
io.ReadAtLeast(0x271f800, 0xc4209eef00, 0xc42296e490, 0x5, 0x5, 0x5, 0xc420cd8180, 0x4, 0x4)
    /usr/local/go/src/io/io.go:307 +0xb2
io.ReadFull(0x271f800, 0xc4209eef00, 0xc42296e490, 0x5, 0x5, 0x5dd412, 0x0, 0x0)
    /usr/local/go/src/io/io.go:325 +0x73
google.golang.org/grpc.(*parser).recvMsg(0xc42296e480, 0x7fffffff, 0x60955c, 0xc420f3d040, 0x0, 0x0, 0x2, 0x6310cc)
    /go/src/google.golang.org/grpc/rpc_util.go:231 +0xa4
--
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc420c52000, 0xc421f16180)
    /go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:187 +0x8b
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
    /go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:188 +0x74

goroutine 19626 [select, 4 minutes]:
google.golang.org/grpc.NewClientStream.func1(0x27347a0, 0xc4224f8700, 0xc422b525a0, 0xc422bd2360)
    /go/src/google.golang.org/grpc/stream.go:187 +0x54a
created by google.golang.org/grpc.NewClientStream
    /go/src/google.golang.org/grpc/stream.go:207 +0xfb9

goroutine 19541 [select, 4 minutes]:
github.com/cockroachdb/cockroach/storage.(*replicaScanner).waitAndProcess(0xc422b64f50, 0xecf343f46, 0xc43075a6eb, 0x2a4ed80, 0xc422f4e040, 0xc420c52120, 0xc421d48900, 0xc422f5b398)
    /go/src/github.com/cockroachdb/cockroach/storage/scanner.go:176 +0x443
github.com/cockroachdb/cockroach/storage.(*replicaScanner).scanLoop.func1.1(0xc421d48900, 0xc422f85020)
    /go/src/github.com/cockroachdb/cockroach/storage/scanner.go:224 +0x124
github.com/cockroachdb/cockroach/storage.(*storeRangeSet).Visit(0xc422f85260, 0xc422ed6e00)
    /go/src/github.com/cockroachdb/cockroach/storage/store.go:274 +0x484
github.com/cockroachdb/cockroach/storage.(*replicaScanner).scanLoop.func1()
    /go/src/github.com/cockroachdb/cockroach/storage/scanner.go:226 +0x350
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc420c52120, 0xc422f37320)
    /go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:187 +0x8b
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
    /go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:188 +0x74

goroutine 19545 [select, 4 minutes]:
github.com/cockroachdb/cockroach/storage.(*baseQueue).processLoop.func1()
    /go/src/github.com/cockroachdb/cockroach/storage/queue.go:402 +0x366
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc420c52120, 0xc422f373e0)
    /go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:187 +0x8b
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
    /go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:188 +0x74

goroutine 19951 [select]:
github.com/cockroachdb/cockroach/storage.(*RaftTransport).processQueue(0xc422df7d40, 0xc421c842a0, 0xc4224f5800, 0x0, 0x0)
    /go/src/github.com/cockroachdb/cockroach/storage/raft_transport.go:269 +0x769
github.com/cockroachdb/cockroach/storage.RaftSender.SendAsync.func1.1()
    /go/src/github.com/cockroachdb/cockroach/storage/raft_transport.go:356 +0xa0
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc420c52000, 0xc421c84360)
    /go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:187 +0x8b
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
    /go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:188 +0x74

goroutine 19670 [select, 4 minutes]:
github.com/cockroachdb/cockroach/storage.(*baseQueue).processLoop.func1()
    /go/src/github.com/cockroachdb/cockroach/storage/queue.go:402 +0x366
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc420c52ab0, 0xc422e8ab80)
    /go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:187 +0x8b
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
    /go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:188 +0x74

goroutine 19615 [select]:
github.com/cockroachdb/cockroach/kv.(*TxnCoordSender).startStats(0xc422e1a6c0)
    /go/src/github.com/cockroachdb/cockroach/kv/txn_coord_sender.go:203 +0xe94
github.com/cockroachdb/cockroach/kv.(*TxnCoordSender).(github.com/cockroachdb/cockroach/kv.startStats)-fm()
    /go/src/github.com/cockroachdb/cockroach/kv/txn_coord_sender.go:188 +0x42
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc420c52ab0, 0xc422cf4be0)
    /go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:187 +0x8b
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
    /go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:188 +0x74

goroutine 19647 [select, 4 minutes]:
google.golang.org/grpc/transport.(*http2Client).controller(0xc4224f8e00)
    /go/src/google.golang.org/grpc/transport/http2_client.go:967 +0x8c9
created by google.golang.org/grpc/transport.newHTTP2Client
    /go/src/google.golang.org/grpc/transport/http2_client.go:201 +0x1381

goroutine 19720 [IO wait]:
net.runtime_pollWait(0x7f8a41529800, 0x72, 0x5de873)
    /usr/local/go/src/runtime/netpoll.go:160 +0x5e
net.(*pollDesc).wait(0xc42192c1b0, 0x72, 0x5db128, 0x1ab1240)
    /usr/local/go/src/net/fd_poll_runtime.go:73 +0x5b
net.(*pollDesc).waitRead(0xc42192c1b0, 0x2723240, 0xc420012098)
    /usr/local/go/src/net/fd_poll_runtime.go:78 +0x42
net.(*netFD).Read(0xc42192c150, 0xc422d8c000, 0x8000, 0x8000, 0x0, 0x2723240, 0xc420012098)
    /usr/local/go/src/net/fd_unix.go:220 +0x1d9
net.(*conn).Read(0xc42012c728, 0xc422d8c000, 0x8000, 0x8000, 0x3e, 0x0, 0x0)
    /usr/local/go/src/net/net.go:173 +0x97
--
google.golang.org/grpc.(*Server).handleRawConn(0xc4224dd040, 0x2734c80, 0xc42012c728)
    /go/src/google.golang.org/grpc/server.go:385 +0x54d
created by google.golang.org/grpc.(*Server).Serve
    /go/src/google.golang.org/grpc/server.go:357 +0x20a

goroutine 19659 [select, 4 minutes]:
github.com/cockroachdb/cockroach/storage.(*baseQueue).processLoop.func1()
    /go/src/github.com/cockroachdb/cockroach/storage/queue.go:402 +0x366
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc420c52750, 0xc422e8a6e0)
    /go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:187 +0x8b
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
    /go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:188 +0x74

goroutine 19594 [select]:
github.com/cockroachdb/cockroach/kv.(*TxnCoordSender).startStats(0xc422e93bc0)
    /go/src/github.com/cockroachdb/cockroach/kv/txn_coord_sender.go:203 +0xe94
github.com/cockroachdb/cockroach/kv.(*TxnCoordSender).(github.com/cockroachdb/cockroach/kv.startStats)-fm()
    /go/src/github.com/cockroachdb/cockroach/kv/txn_coord_sender.go:188 +0x42
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc420c52750, 0xc4224d17e0)
    /go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:187 +0x8b
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
    /go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:188 +0x74

goroutine 19609 [select]:
github.com/cockroachdb/cockroach/gossip.(*client).gossip(0xc422a17930, 0x7f8a4156ef30, 0xc422df7880, 0xc4216d6580, 0x271c2c0, 0xc42012c410, 0xc420c52000, 0x0, 0x0)
    /go/src/github.com/cockroachdb/cockroach/gossip/client.go:278 +0x6df
github.com/cockroachdb/cockroach/gossip.(*client).start.func1()
    /go/src/github.com/cockroachdb/cockroach/gossip/client.go:93 +0x2e9
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc420c52000, 0xc422df77c0)
    /go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:187 +0x8b
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
    /go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:188 +0x74

goroutine 19652 [select, 4 minutes]:
github.com/cockroachdb/cockroach/storage.(*baseQueue).processLoop.func1()
    /go/src/github.com/cockroachdb/cockroach/storage/queue.go:402 +0x366
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc420c52750, 0xc422e8a5c0)
    /go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:187 +0x8b
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
    /go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:188 +0x74

goroutine 19532 [select]:
github.com/cockroachdb/cockroach/storage.(*Store).startGossip.func2()
    /go/src/github.com/cockroachdb/cockroach/storage/store.go:1227 +0x2e9
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc420c52120, 0xc422f93d40)
    /go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:187 +0x8b
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
    /go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:188 +0x74

goroutine 19504 [select, 4 minutes]:
github.com/cockroachdb/cockroach/storage.(*StorePool).start.func1()
    /go/src/github.com/cockroachdb/cockroach/storage/store_pool.go:308 +0x519
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc420c52120, 0xc422f92980)
    /go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:187 +0x8b
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
    /go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:188 +0x74

goroutine 19974 [select, 4 minutes]:
github.com/cockroachdb/cockroach/storage.(*RaftTransport).RaftMessage(0xc422f4f6c0, 0x27345c0, 0xc421928370, 0xc421bc2ba8, 0x626e80)
    /go/src/github.com/cockroachdb/cockroach/storage/raft_transport.go:156 +0x35e
github.com/cockroachdb/cockroach/storage._MultiRaft_RaftMessage_Handler(0x1b2db60, 0xc422f4f6c0, 0x27309c0, 0xc420f3d170, 0x33, 0x0)
    /go/src/github.com/cockroachdb/cockroach/storage/raft.pb.go:158 +0xda
google.golang.org/grpc.(*Server).processStreamingRPC(0xc4224dc820, 0x2734800, 0xc4228e7290, 0xc4209ef3b0, 0xc422f85f80, 0x2706d60, 0xc421f16b70, 0x0, 0x0)
    /go/src/google.golang.org/grpc/server.go:686 +0xc7d
google.golang.org/grpc.(*Server).handleStream(0xc4224dc820, 0x2734800, 0xc4228e7290, 0xc4209ef3b0, 0xc421f16b70)
    /go/src/google.golang.org/grpc/server.go:770 +0x170e
google.golang.org/grpc.(*Server).serveStreams.func1.1(0xc421f8a8e0, 0xc4224dc820, 0x2734800, 0xc4228e7290, 0xc4209ef3b0)
    /go/src/google.golang.org/grpc/server.go:419 +0xb9
created by google.golang.org/grpc.(*Server).serveStreams.func1
    /go/src/google.golang.org/grpc/server.go:420 +0xbb

goroutine 19603 [select, 4 minutes]:
github.com/cockroachdb/cockroach/storage.(*Store).Start.func4()
    /go/src/github.com/cockroachdb/cockroach/storage/store.go:1134 +0x19d
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc420c52750, 0xc422ecc4e0)
    /go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:187 +0x8b
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
    /go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:188 +0x74

goroutine 19575 [select]:
google.golang.org/grpc/transport.(*recvBufferReader).Read(0xc422e5e300, 0xc422f37b30, 0x5, 0x5, 0x0, 0x0, 0x0)
    /go/src/google.golang.org/grpc/transport/transport.go:141 +0x898
google.golang.org/grpc/transport.(*Stream).Read(0xc420cf6ff0, 0xc422f37b30, 0x5, 0x5, 0xc422e5e3c0, 0xc420fcbad0, 0x626fa0)
    /go/src/google.golang.org/grpc/transport/transport.go:314 +0x79
io.ReadAtLeast(0x271f800, 0xc420cf6ff0, 0xc422f37b30, 0x5, 0x5, 0x5, 0x5de873, 0xc4209bc3e8, 0xc420fcbc3c)
    /usr/local/go/src/io/io.go:307 +0xb2
io.ReadFull(0x271f800, 0xc420cf6ff0, 0xc422f37b30, 0x5, 0x5, 0x6093d3, 0x15fd720, 0x7f8a2cd2e948)
    /usr/local/go/src/io/io.go:325 +0x73
google.golang.org/grpc.(*parser).recvMsg(0xc422f37b20, 0x7fffffff, 0xc420fcbea0, 0xb9464a, 0xc422a16dd8, 0xc422d8a090, 0x1, 0x340cec0)
    /go/src/google.golang.org/grpc/rpc_util.go:231 +0xa4
--
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc420c52000, 0xc422f572c0)
    /go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:187 +0x8b
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
    /go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:188 +0x74

goroutine 19618 [chan receive, 4 minutes]:
github.com/cockroachdb/cockroach/gossip.(*server).start.func3()
    /go/src/github.com/cockroachdb/cockroach/gossip/server.go:323 +0x92
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc420c52000, 0xc422ecd860)
    /go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:187 +0x8b
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
    /go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:188 +0x74

goroutine 19721 [select]:
google.golang.org/grpc/transport.(*http2Server).controller(0xc420c53b00)
    /go/src/google.golang.org/grpc/transport/http2_server.go:673 +0xa56
created by google.golang.org/grpc/transport.newHTTP2Server
    /go/src/google.golang.org/grpc/transport/http2_server.go:139 +0x9b5

goroutine 19562 [select]:
github.com/cockroachdb/cockroach/storage.(*Store).startGossip.func1()
    /go/src/github.com/cockroachdb/cockroach/storage/store.go:1208 +0x2e9
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc420c52480, 0xc422eb1040)
    /go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:187 +0x8b
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
    /go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:188 +0x74

goroutine 19605 [select]:
github.com/cockroachdb/cockroach/storage.(*Store).startGossip.func2()
    /go/src/github.com/cockroachdb/cockroach/storage/store.go:1227 +0x2e9
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc420c52750, 0xc422ecc580)
    /go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:187 +0x8b
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
    /go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:188 +0x74

goroutine 19957 [select, 4 minutes]:
google.golang.org/grpc/transport.(*recvBufferReader).Read(0xc421c7ee00, 0xc4229bb2d0, 0x5, 0x5, 0x0, 0x0, 0x0)
    /go/src/google.golang.org/grpc/transport/transport.go:141 +0x898
google.golang.org/grpc/transport.(*Stream).Read(0xc4216faff0, 0xc4229bb2d0, 0x5, 0x5, 0xc4229ba630, 0x5, 0x5)
    /go/src/google.golang.org/grpc/transport/transport.go:314 +0x79
io.ReadAtLeast(0x271f800, 0xc4216faff0, 0xc4229bb2d0, 0x5, 0x5, 0x5, 0x5, 0x5, 0x5)
    /usr/local/go/src/io/io.go:307 +0xb2
io.ReadFull(0x271f800, 0xc4216faff0, 0xc4229bb2d0, 0x5, 0x5, 0xc4229ba630, 0x5, 0x5)
    /usr/local/go/src/io/io.go:325 +0x73
google.golang.org/grpc.(*parser).recvMsg(0xc4229bb2c0, 0x7fffffff, 0xc4315e, 0xc4229ba620, 0x7fffffff, 0xc4224dd060, 0x0, 0x0)
    /go/src/google.golang.org/grpc/rpc_util.go:231 +0xa4
--
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc420c52000, 0xc4229bb2e0)
    /go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:187 +0x8b
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
    /go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:188 +0x74

goroutine 19634 [select]:
github.com/cockroachdb/cockroach/gossip.(*server).Gossip(0xc422a16410, 0x27343e0, 0xc422dfba60, 0x0, 0x0)
    /go/src/github.com/cockroachdb/cockroach/gossip/server.go:157 +0x73b
github.com/cockroachdb/cockroach/gossip._Gossip_Gossip_Handler(0x1af3e00, 0xc422a16410, 0x27309c0, 0xc4228e6900, 0x0, 0x0)
    /go/src/github.com/cockroachdb/cockroach/gossip/gossip.pb.go:211 +0xda
google.golang.org/grpc.(*Server).processStreamingRPC(0xc4224dc000, 0x2734800, 0xc4228e6630, 0xc420cf73b0, 0xc422f84b40, 0x2706c00, 0xc4209bc780, 0x0, 0x0)
    /go/src/google.golang.org/grpc/server.go:686 +0xc7d
google.golang.org/grpc.(*Server).handleStream(0xc4224dc000, 0x2734800, 0xc4228e6630, 0xc420cf73b0, 0xc4209bc780)
    /go/src/google.golang.org/grpc/server.go:770 +0x170e
google.golang.org/grpc.(*Server).serveStreams.func1.1(0xc422dfb2a0, 0xc4224dc000, 0x2734800, 0xc4228e6630, 0xc420cf73b0)
    /go/src/google.golang.org/grpc/server.go:419 +0xb9
created by google.golang.org/grpc.(*Server).serveStreams.func1
    /go/src/google.golang.org/grpc/server.go:420 +0xbb

goroutine 19573 [select, 4 minutes]:
google.golang.org/grpc/transport.(*http2Client).controller(0xc4224f8700)
    /go/src/google.golang.org/grpc/transport/http2_client.go:967 +0x8c9
created by google.golang.org/grpc/transport.newHTTP2Client
    /go/src/google.golang.org/grpc/transport/http2_client.go:201 +0x1381

goroutine 19646 [IO wait]:
net.runtime_pollWait(0x7f8a41528fc0, 0x72, 0x5de873)
    /usr/local/go/src/runtime/netpoll.go:160 +0x5e
net.(*pollDesc).wait(0xc421d3a610, 0x72, 0x5db128, 0x1ab1240)
    /usr/local/go/src/net/fd_poll_runtime.go:73 +0x5b
net.(*pollDesc).waitRead(0xc421d3a610, 0x2723240, 0xc420012098)
    /usr/local/go/src/net/fd_poll_runtime.go:78 +0x42
net.(*netFD).Read(0xc421d3a5b0, 0xc4207b6000, 0x8000, 0x8000, 0x0, 0x2723240, 0xc420012098)
    /usr/local/go/src/net/fd_unix.go:220 +0x1d9
net.(*conn).Read(0xc420028760, 0xc4207b6000, 0x8000, 0x8000, 0x2, 0x0, 0x0)
    /usr/local/go/src/net/net.go:173 +0x97
--
google.golang.org/grpc/transport.(*http2Client).reader(0xc4224f8e00)
    /go/src/google.golang.org/grpc/transport/http2_client.go:889 +0x157
created by google.golang.org/grpc/transport.newHTTP2Client
    /go/src/google.golang.org/grpc/transport/http2_client.go:171 +0xbaf

goroutine 19567 [select, 4 minutes]:
github.com/cockroachdb/cockroach/storage.(*baseQueue).processLoop.func1()
    /go/src/github.com/cockroachdb/cockroach/storage/queue.go:402 +0x366
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc420c52480, 0xc422eb1100)
    /go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:187 +0x8b
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
    /go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:188 +0x74

goroutine 19637 [select]:
github.com/cockroachdb/cockroach/storage.(*Store).startGossip.func1()
    /go/src/github.com/cockroachdb/cockroach/storage/store.go:1208 +0x2e9
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc420c52ab0, 0xc422ed2b60)
    /go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:187 +0x8b
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
    /go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:188 +0x74

goroutine 19960 [select]:
google.golang.org/grpc/transport.(*recvBufferReader).Read(0xc421c1d5c0, 0xc4229bb510, 0x5, 0x5, 0x0, 0x0, 0x0)
    /go/src/google.golang.org/grpc/transport/transport.go:141 +0x898
google.golang.org/grpc/transport.(*Stream).Read(0xc4209ef4a0, 0xc4229bb510, 0x5, 0x5, 0x0, 0xc420cb6400, 0xc420cb63c0)
    /go/src/google.golang.org/grpc/transport/transport.go:314 +0x79
io.ReadAtLeast(0x271f800, 0xc4209ef4a0, 0xc4229bb510, 0x5, 0x5, 0x5, 0xc420cb6400, 0x4, 0x4)
    /usr/local/go/src/io/io.go:307 +0xb2
io.ReadFull(0x271f800, 0xc4209ef4a0, 0xc4229bb510, 0x5, 0x5, 0x30, 0x0, 0x0)
    /usr/local/go/src/io/io.go:325 +0x73
google.golang.org/grpc.(*parser).recvMsg(0xc4229bb500, 0x7fffffff, 0x60955c, 0xc4205b8470, 0x0, 0x0, 0x2, 0x6310cc)
    /go/src/google.golang.org/grpc/rpc_util.go:231 +0xa4
--
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc420c52000, 0xc420dca300)
    /go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:187 +0x8b
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
    /go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:188 +0x74

goroutine 19674 [select, 4 minutes]:
github.com/cockroachdb/cockroach/storage.(*baseQueue).processLoop.func1()
    /go/src/github.com/cockroachdb/cockroach/storage/queue.go:402 +0x366
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc420c52ab0, 0xc422e8ac20)
    /go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:187 +0x8b
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
    /go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:188 +0x74

goroutine 19617 [IO wait, 4 minutes]:
net.runtime_pollWait(0x7f8a41529140, 0x72, 0x5de873)
    /usr/local/go/src/runtime/netpoll.go:160 +0x5e
net.(*pollDesc).wait(0xc421d00e60, 0x72, 0x5db128, 0x1ab1240)
    /usr/local/go/src/net/fd_poll_runtime.go:73 +0x5b
net.(*pollDesc).waitRead(0xc421d00e60, 0x2723240, 0xc420012098)
    /usr/local/go/src/net/fd_poll_runtime.go:78 +0x42
net.(*netFD).accept(0xc421d00e00, 0x0, 0x2720700, 0xc422ca0ae0)
    /usr/local/go/src/net/fd_unix.go:396 +0x2b8
net.(*TCPListener).accept(0xc42012c510, 0xc420028ac0, 0xc421bd4e98, 0x6035af)
    /usr/local/go/src/net/tcpsock_posix.go:132 +0x51
--
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc420c52000, 0xc422ecd800)
    /go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:187 +0x8b
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
    /go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:188 +0x74

goroutine 19536 [chan receive, 4 minutes]:
github.com/cockroachdb/cockroach/storage/engine.(*RocksDB).Open.func1(0xc422f0a780)
    /go/src/github.com/cockroachdb/cockroach/storage/engine/rocksdb.go:384 +0x64
created by github.com/cockroachdb/cockroach/storage/engine.(*RocksDB).Open
    /go/src/github.com/cockroachdb/cockroach/storage/engine/rocksdb.go:385 +0xb59

goroutine 19604 [select]:
github.com/cockroachdb/cockroach/storage.(*Store).startGossip.func1()
    /go/src/github.com/cockroachdb/cockroach/storage/store.go:1208 +0x2e9
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc420c52750, 0xc422ecc500)
    /go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:187 +0x8b
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
    /go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:188 +0x74

goroutine 19958 [select, 4 minutes]:
google.golang.org/grpc.NewClientStream.func1(0x27347a0, 0xc4224f9d00, 0xc4216fb0e0, 0xc4218bbe60)
    /go/src/google.golang.org/grpc/stream.go:187 +0x54a
created by google.golang.org/grpc.NewClientStream
    /go/src/google.golang.org/grpc/stream.go:207 +0xfb9

goroutine 19527 [select]:
github.com/cockroachdb/cockroach/storage.(*bookie).start.func1()
    /go/src/github.com/cockroachdb/cockroach/storage/reservation.go:275 +0x4c8
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc420c52120, 0xc422f937c0)
    /go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:187 +0x8b
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
    /go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:188 +0x74

goroutine 19590 [select, 4 minutes]:
github.com/cockroachdb/cockroach/storage.(*replicaScanner).waitAndProcess(0xc4213585b0, 0xecf343f46, 0xc431ae1d03, 0x2a4ed80, 0xc422f4e040, 0xc420c52480, 0x0, 0xc422eb1040)
    /go/src/github.com/cockroachdb/cockroach/storage/scanner.go:176 +0x443
github.com/cockroachdb/cockroach/storage.(*replicaScanner).scanLoop.func1()
    /go/src/github.com/cockroachdb/cockroach/storage/scanner.go:229 +0x4eb
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc420c52480, 0xc422eb12c0)
    /go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:187 +0x8b
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
    /go/src/github.com/cockroachdb/cockroac
```

Please assign, take a look and update the issue accordingly.

--- a/pkg/storage/client_raft_test.go
+++ b/pkg/storage/client_raft_test.go
@@ -42,6 +42,7 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/util"
 	"github.com/cockroachdb/cockroach/pkg/util/hlc"
 	"github.com/cockroachdb/cockroach/pkg/util/leaktest"
+	"github.com/cockroachdb/cockroach/pkg/util/log"
 	"github.com/cockroachdb/cockroach/pkg/util/stop"
 	"github.com/cockroachdb/cockroach/pkg/util/syncutil"
 	"github.com/cockroachdb/cockroach/pkg/util/timeutil"
@@ -2195,12 +2196,33 @@ func TestReplicateRogueRemovedNode(t *testing.T) {
 	finishWG.Wait()
 }
 
+type disruptiveElectionTestHandler chan *roachpb.Error
+
+func (disruptiveElectionTestHandler) HandleRaftRequest(
+	_ context.Context, _ *storage.RaftMessageRequest, _ storage.RaftMessageResponseStream,
+) *roachpb.Error {
+	panic("unimplemented")
+}
+
+func (d disruptiveElectionTestHandler) HandleRaftResponse(
+	ctx context.Context, resp *storage.RaftMessageResponse,
+) {
+	switch val := resp.Union.GetValue().(type) {
+	case *roachpb.Error:
+		d <- val
+	default:
+		log.Fatalf(ctx, "unexpected response type %T", val)
+	}
+}
+
+func (disruptiveElectionTestHandler) HandleSnapshot(
+	_ *storage.SnapshotRequest_Header, _ storage.SnapshotResponseStream,
+) error {
+	panic("unimplemented")
+}
+
 func TestReplicateRemovedNodeDisruptiveElection(t *testing.T) {
 	defer leaktest.AfterTest(t)()
-	// The change to error reporting means that we can no longer trap
-	// transport errors separately from error messages and send them to
-	// errChan.
-	t.Skip("TODO(bdarnell): flaky (#8308), and needs update for change to raft transport error reporting")
 
 	mtc := startMultiTestContext(t, 4)
 	defer mtc.Stop()
@@ -2211,12 +2233,42 @@ func TestReplicateRemovedNodeDisruptiveElection(t *testing.T) {
 	mtc.unreplicateRange(rangeID, 0)
 	mtc.expireLeases()
 
-	// Write on the second node, to ensure that the other nodes have
-	// established a lease after the first node's removal.
-	incArgs := incrementArgs([]byte("a"), 5)
+	// Ensure that we have a stable lease and raft leader so we can tell if the
+	// removed node causes a disruption. This is a three-step process.
+
+	// 1. Write on the second node, to ensure that a lease has been
+	// established after the first node's removal.
+	key := roachpb.Key("a")
+	value := int64(5)
+	incArgs := incrementArgs(key, value)
 	if _, err := client.SendWrapped(context.Background(), mtc.distSenders[1], &incArgs); err != nil {
 		t.Fatal(err)
 	}
+
+	// 2. Wait for all nodes to process the increment (and therefore the
+	// new lease).
+	mtc.waitForValues(key, []int64{0, value, value, value})
+
+	// 3. Wait for the lease holder to obtain raft leadership too.
+	util.SucceedsSoon(t, func() error {
+		req := &roachpb.LeaseInfoRequest{
+			Span: roachpb.Span{
+				Key: roachpb.KeyMin,
+			},
+		}
+		reply, pErr := client.SendWrapped(context.Background(), mtc.distSenders[1], req)
+		if pErr != nil {
+			return pErr.GoError()
+		}
+		leaseReplica := reply.(*roachpb.LeaseInfoResponse).Lease.Replica.ReplicaID
+		leadReplica := roachpb.ReplicaID(mtc.stores[1].RaftStatus(rangeID).Lead)
+		if leaseReplica != leadReplica {
+			return errors.Errorf("leaseReplica %s does not match leadReplica %s",
+				leaseReplica, leadReplica)
+		}
+
+		return nil
+	})
 
 	// Save the current term, which is the latest among the live stores.
 	findTerm := func() uint64 {
@@ -2234,6 +2286,8 @@ func TestReplicateRemovedNodeDisruptiveElection(t *testing.T) {
 		t.Fatalf("expected non-zero term")
 	}
 
+	// replica0 is the one that  has been removed; replica1 is a current
+	// member of the group.
 	replica0 := roachpb.ReplicaDescriptor{
 		ReplicaID: roachpb.ReplicaID(mtc.stores[0].StoreID()),
 		NodeID:    roachpb.NodeID(mtc.stores[0].StoreID()),
@@ -2244,9 +2298,20 @@ func TestReplicateRemovedNodeDisruptiveElection(t *testing.T) {
 		NodeID:    roachpb.NodeID(mtc.stores[1].StoreID()),
 		StoreID:   mtc.stores[1].StoreID(),
 	}
+
+	// Create a new transport for store 0. Error responses are passed
+	// back along the same grpc stream as the request so it's ok that
+	// there are two (this one and the one actually used by the store).
+	transport0 := storage.NewRaftTransport(log.AmbientContext{},
+		storage.GossipAddressResolver(mtc.gossips[0]),
+		nil, /* grpcServer */
+		mtc.rpcContext,
+	)
+	errChan := disruptiveElectionTestHandler(make(chan *roachpb.Error, 1))
+	transport0.Listen(mtc.stores[0].StoreID(), errChan)
+
 	// Simulate an election triggered by the removed node.
-	errChan := make(chan error)
-	mtc.transports[0].SendAsync(&storage.RaftMessageRequest{
+	transport0.SendAsync(&storage.RaftMessageRequest{
 		RangeID:     rangeID,
 		ToReplica:   replica1,
 		FromReplica: replica0,
@@ -2258,8 +2323,16 @@ func TestReplicateRemovedNodeDisruptiveElection(t *testing.T) {
 		},
 	})
 
-	if err := <-errChan; !testutils.IsError(err, "sender replica too old, discarding message") {
-		t.Fatalf("got unexpected error: %v", err)
+	// The receiver of this message should return an error.
+	select {
+	case pErr := <-errChan:
+		switch pErr.GetDetail().(type) {
+		case *roachpb.ReplicaTooOldError:
+		default:
+			t.Fatalf("unexpected error type %T: %s", pErr, pErr)
+		}
+	case <-time.After(time.Second):
+		t.Fatal("did not get expected error")
 	}
 
 	// The message should have been discarded without triggering an

